pkg/apiserver/authz.go

```diff
@@ -108,3 +115,36 @@ func NewAuthorizerFromAuthorizationConfig(authorizationModes []string, authoriza
 
 	return union.New(authorizers...), nil
 }
+
+func checkAndReloadPolicyFile(policyAuthz *abac.PolicyAuthorizer, file string) {
+	lastModifyTime := time.Now()
+	for {
+		func() {
+			defer func() {
+				if x := recover(); x != nil {
+					glog.Errorf("APIServer panic'd on checkAndReloadPolicyFile: %s, err: %v", file, x)
+				}
+			}()
+			//var info os.FileInfo
+			info, err := os.Stat(file)
+			if err != nil {
+				glog.Errorf("Stat authorizationPolicyFile: %s fail, err: %v, will retry to reload later", file, err)
+				return
+			}
+			if info.ModTime().After(lastModifyTime) {
+				lastModifyTime = info.ModTime()
+				newPolicyList, err := abac.NewFromFile(file)
+				if err != nil { // file format not correct
+					glog.Errorf("Stat authorizationPolicyFile: %s fail, err: %v, will retry to reload later", file, err)
+					return
+				}
+
+				glog.Infof("authorizationPolicyFile: %s is modified, reload finished, number of policy lines change from %d to %d", file, len(*policyAuthz.PL), len(newPolicyList))
+				policyAuthz.Lock.Lock()
+				*(policyAuthz.PL) = newPolicyList
+				policyAuthz.Lock.Unlock()
+			}
+		}()
+		time.Sleep(5 * time.Second)
```

Comment:
It' better to define sleep time as a constant. Such as `const recheckIntervel = 5*time.Second`